components/VM_Arm/src/main.c

```diff
@@ -715,6 +717,34 @@ static int route_irqs(vm_vcpu_t *vcpu, irq_server_t *irq_server)
     return 0;
 }
 
+/* Different from load_image which loads the image into the guest's address space. This pulls the
+ * image from the FileServer and stores it into a local buffer
+ */
+static int get_file(const char *name, char *buf)
+{
+    int fd;
+    size_t len;
+
+    fd = open(name, 0);
+    if (fd == -1) {
+        ZF_LOGE("Error: Unable to open image \'%s\'", name);
+        return 0;
+    }
+
+    size_t offset = 0;
+    while (1) {
+        /* Load the image BIT(20) bytes at a time */
```

Comment:
- Seems this function is a generic file loader after all. So there should also be a max buffer size parameter passed to have user from surprises when the file is really huger.
- Can you add a comment why BIT(20) is the limit here actually? It this a random choice or is this the shared memory limit that must be kept in mind here?

Reply:
BIT(20) is 1MB, and the `get_file` function was pulled from the seL4_projects_libs library, which pulled 1MB at a time. There really isn't a reason.

I'll add a max_size parameter. The library version just loads images to guest address space, so there isn't a worry about buffer overruns.